### ISSUE TITLE
OC Advanced Settings box includes the Coordinator's 'Inventory Setting'

### ISSUE THREAD

**Issue by kirstenalarsen** [openfoodfoundation/openfoodnetwork#5129]
## Description
** As a Hub user I am setting up an order cycle and I want to use Inventory
** I want to be able to see, understand and modify all the Inventory related things affecting which products are in my shop in one place

Related to #5128

Currently the Advanced Settings box contains the Inventory control for this Order Cycle. 

I am proposing that it should also contain 
** The current setting for whether 'New Products' must be added to Inventory or are available as is, including notification that changing this will change it for all Order Cycles not just this one

![image.png](https://images.zenhubusercontent.com/54b83fb37c6ccc6277a6ea46/1f54e49c-f034-4166-a896-3e162aa27edf)

** A red notification if there are products that are hidden from the Coordinators Inventory that  cannot be included in the Outgoing Shop until they are placed in Inventory, regardless of the settings above (I think doing this would also alleviate #4804) - with a link to the hidden inventory list for that Coordinator, that opens in another tab.

## Acceptance Criteria & Tests

1. #5128 Dependency - User can modify Inventory settings from each tab in the Order Cycle
2.  User can identify and modify (or at least find) all relevant Inventory settings that are affecting which products are showing in their shop

**Comment by luisramos0** [comment]
:+1:
I think this needs some inception first: "with a link to the hidden inventory list for that Coordinator, that opens in another tab."

**Comment by tschumilas** [comment]
It would be very helpful to have the inventory settings moved from the profile menu,  to the OC advanced settings.  All in one place would help users who are playing with this.   I've had a couple of user problems where they misunderstood the impact of the inventory settings in profile,  and then couldn't figure out why things weren't or were in their shop.   If this is all in 'advanced' settings - it says to users,  don't start mucking with me first.

**Comment by luisramos0** [comment]
I am not sure we can move the profile one to the OC page, they are different things.

there are two preferences now:
- the "Enterprise preference" on the enterprise profile
- the "Order Cycle preference" in the order cycle page

I found the slack posts I wrote after reading the code about this:
- the Enterprise preference is to restrict the catalog to the inventory for that enterprise **in any OC**, the OC preference is to restrict the products in that OC to the coordinator’s inventory.
- **the Enterprise preference is used on outgoing exchanges and the OC preference is used on incoming exchanges**
- the OC preference restricts incoming products to the coordinator inventory while the Enterprise preference restricts outgoing products to the inventory of the enterprise selling them

I hope this is useful for the discussion.

**Comment by luisramos0** [comment]
starting the inception for ""with a link to the hidden inventory list for that Coordinator, that opens in another tab."
I think this list is exactly what's seen on the inventory management page for the coordinator if then go to "new products", right? all the products they can see that are not yet on inventory. All these will be "hidden" if the OC preference is turned on.

**Comment by kirstenalarsen** [comment]
No I don't think so @luisramos0 
- there are three lists in Inventory - those I have put in my inventory, those I have actively hidden and anything in my 'review' list i.e. that I have not actively included NOR hidden
![image](https://user-images.githubusercontent.com/2911176/78539501-05420000-7836-11ea-8fc8-b857cefc8536.png)

It seems to me that the Coordinators' setting here is important because in MOST cases (although you are correct, not all) the Coordinator is also the outgoing exchange. So if this setting is set to 'include new products' then all the products that are available but they have not reviewed are available in incoming. But if the setting is set to 'must add to inventory' then then don't see the new products. So it's pretty damn confusing

But you are right I think (haven't tested to demonstrate to myself) about the outgoing exchanges - so that would add potentially another layer. I am a bit brain-exploded about what would happen if outgoing hubs have different settings on this - because the products will only be available to the order cycle according to the coord setting. But then they also need to be according to the outgoing enterprise setting - what if the coordinator doesn't manage that inventory? they can't put the products in the OC? I am very confused!! 

Still think that displaying the Coordinator setting here would be useful - especially in the Incoming page because it does control what you can see there

**Comment by luisramos0** [comment]
-- Still think that displaying the Coordinator setting here would be useful - especially in the Incoming page because it does control what you can see there

Believing in what I have found in the code some months ago, this is not correct: the Coordinator setting only controls what they see in **their enterprise outgoing exchange**.

Shall I dive in the code again to confirm?

So, adding the enterprise setting in the OC page is incorrect unless it is added to each outgoing exchange in the outgoing exchanges page.

**Comment by tschumilas** [comment]
I love this inventory stuff.  I want to dig in -  will do some testing and share info later this morning -  I just HAVE to do some screenshots to 'Canadianze'  the AUS farmers market post --- which is awesome - first.    A few months back I ran tests - and had a 2-way chart with the 2 inventory control conditions (Inventory settings  vs  Advanced Settings),  and I had this for different enterprise types.  BUT - it was written on a piece of paper and I lost it!  Sheesh.  Will re-do.  But i remember thinking at the time that,  if the Inventory setting is set that products must be added to inventory,  then then the advanced OC setting does nothing because show all products = shows only inventory products.   Products in product lists -  never put into inventory - never show in the OC.   this confuses people -  because they are assuming 'show all products' shows all products   --- but if I remember my test correctly ,  it only shows ALL products if the profile inventory setting is set to all products can be added.   Need to re-do chart.

**Comment by kirstenalarsen** [comment]
yes i would love to see that chart!! If we can't agree how it works it's no wonder people can't use it!!

**Comment by luisramos0** [comment]
" if the Inventory setting is set that products must be added to inventory, then then the advanced OC setting does nothing"

@tschumilas I think your explanation is correct in terms of what gets to thee shopfront BUT I think you are missing the difference between incoming and outgoing that I mentioned.
In this case where the enterprise setting is inventory only:
- with OC setting OFF/catalog, the coordinator will see ALL products in the incoming exchange but they will not flow to their outgoing exchange (because of the enterprise level preference) 
- with OC setting ON/inventory,  the coordinator will only see inventory in incoming and outgoing

makes sense?

**Comment by Erioldoesdesign** [comment]
This feels 'not papercut-y' and feels more like a piece where I, as a user centred designer, want to fully understand the scenarios in which inventory is used, by hubs, across how many producers/enterprises and what are the key needs to edit, modify, add and work across multiple order cycles and potentially, from a producer pov, across multiple hub co-ordinators that I potentially supply.

I *think* I understand the basic need which is for hub co-ordinators 'to see more' in the order cycle section which is a very reasonable need. What I don't know is reasonable is to be able to fully manage and edit 'everything' from that place when it impacts (what I'm assuming) is other producers, hubs etc. 

I wonder if, and this will be a bit abstract, there is a a way to allow hub-coordinator to see more but not edit? (this might already be the case with hidden products but I don't think it is)

**Comment by luisramos0** [comment]
From my comments above (I have spent some good time reading the code), I sound pretty sure **the Enterprise settings only influence the outgoing exchanges of that enterprises (unrelated to the coordinator)** so the small improvement I can see we could make would be to **add this enterprise setting to each outgoing exchange** (somewhere on the line or within the tab where the products are listed).
In this example below: bring and show the "Enterprise Inventory Only setting" to this page for each of these enterprises with outgoing exchanges. In this case for example, Fredo's and Mary's may have different settings. We could show the setting as a label "Only Products in Inventory are included in this exchange, change Freddo's enterprise settings if you want to include catalog products as well" that would clearly show that for example Freddo's settings is filtering out products that are not in inventory:
![image](https://user-images.githubusercontent.com/1640378/98032423-e88d9b00-1e0b-11eb-9d87-c7fba0ae01f2.png)

@Erioldoesdesign I wonder how we should handle this in terms of process. If there's no consensus about my suggestion above, this is not a papercut anymore as you suggest. Correct? Maybe we move this issue to the product and design repo?
